lib/maths/unittest/CBoostedTreeTest.cc

```diff
@@ -109,14 +114,17 @@ auto predictionStatistics(test::CRandomNumbers& rng,
             double predictionErrorVariance{
                 maths::CBasicStatistics::variance(modelPredictionErrorMoments)};
 
-            modelPredictionBias.push_back(predictionErrorMean);
-            modelPredictionMseImprovement.push_back(
-                (functionVariance - noiseVariance / static_cast<double>(rows)) /
-                (predictionErrorVariance - noiseVariance / static_cast<double>(rows)));
+            modelBias.push_back(predictionErrorMean);
+            modelRSquared.push_back(
+                1.0 -
+                (predictionErrorVariance - noiseVariance / static_cast<double>(rows)) /
+                    (functionVariance - noiseVariance / static_cast<double>(rows)));
         }
     }
+    LOG_DEBUG(<< "bias = " << core::CContainerPrinter::print(modelBias));
+    LOG_DEBUG(<< " R^2 = " << core::CContainerPrinter::print(modelRSquared));
```

Comment:
nit: For me it would be rather an unusual combination of KPIs. Why not using MSE or RMSE for regression? Or explained variance instead of R^2?

Reply:
I moved away from a non-normalised because I felt it was more useful for reading test assertions at a glance. I feel like checking for bias on average is interesting since the threshold is significantly tighter.

Wouldn't you define explained R^2 as explained variance since \sum_i{(\hat{y_i}-\bar{y})^2} = \sum_i{(y_i-\bar{y})^2}-\sum_i{(y_i-\hat{y_i})^2}?

Reply:
Oh sorry, I get what you mean just (non-relative) explained variance.

Reply:
Ok, so I think R^2 is the natural generalisation of explained variance for general models (coincides for OLS, but not in general). I want to keep a separate assessment of the *average* bias because this should be significantly smaller than the MSE. I also want a normalised measure. Whilst I could base this on say MSE I think R^2 is more standard. Therefore, I think on balance I'll keep this as I've changed it.